### PR TITLE
feat: Admission Controller in OpenShift

### PR DESCRIPTION
**Pull request by rollandf**
- Add 'openshift' kustomize dir under config/manifests
- Add Admission Controller yamls to CSV
- Fix Mellanox CRD finalizers permission
- Add Env variable to skip validations if needed

### CODE REVIEW

**Comment by rollandf** [comment]
Note that `relatedImages` are deleted when running `make bundle`.
For now I have updated the CSV manually.
I have opened an issue in operator-sdk: https://github.com/operator-framework/operator-sdk/issues/6590

File: config/manifests/openshift/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+bases:
+- ../../crd
+- ../../rbac
+- ../../manager
+- ../../webhook
+
+patchesStrategicMerge:
+- manager_webhook_patch.yaml
+- webhookservicecainjection_patch.yaml
```
Comment (adrianchiris) [inline config/manifests/openshift/kustomization.yaml:19]: nit:newline

File: docs/operator-bundle.md [hunk 2]
```diff
@@ -81,3 +81,26 @@ To remove the operator when installed via `operator-sdk run`, use:
...
+
+Get the `Subscription` name:
+
+```
+kubectl get subscriptions.operators.coreos.com -n nvidia-network-operator
+NAME                                  PACKAGE                   SOURCE                            CHANNEL
+nvidia-network-operator-v23-7-0-sub   nvidia-network-operator   nvidia-network-operator-catalog   v23.7.0
+```
+
+Edit the `Subscription`, and add a section `config\env` under `spec` with needed vars and values.
```
Comment (adrianchiris) [inline docs/operator-bundle.md:97]: nit: `config/env` -> `spec.config.env` ?

File: bundle/manifests/nvidia-network-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -645,15 +655,54 @@ spec:
...
   - image: ghcr.io/k8snetworkplumbingwg/sriov-network-device-plugin@sha256:f717f9778f48665b7c592f2225df51b755a1fe048125e034a286c564ee10fd37
     name: sriov-network-device-plugin
   - image: nvcr.io/nvidia/cloud-native/k8s-rdma-shared-dev-plugin@sha256:941ad9ff5013e9e7ad5abeb0ea9f79d45379cfae88a628d923f87d2259bdd132
     name: rdma-shared-device-plugin
-  - image: gcr.io/kubebuilder/kube-rbac-proxy@sha256:db06cc4c084dd0253134f156dddaaf53ef1c3fb3cc809e5d81711baa4029ea4c
-    name: kube-rbac-proxy
   - image: nvcr.io/nvidia/cloud-native/network-operator@sha256:7005fa24a1ae52d927e76d50d90fddf6b6c7b08885a2dad3c7e5e2c2ac21c834
-    name: network-operator
-  version: 23.7.0
+    name: network-operator
```
Comment (adrianchiris) [inline bundle/manifests/nvidia-network-operator.clusterserviceversion.yaml:708]: nit:  newline

File: docs/operator-bundle.md (hunk 2 above)
Comment (rollandf) [reply docs/operator-bundle.md]: Done

File: bundle/manifests/nvidia-network-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (rollandf) [reply bundle/manifests/nvidia-network-operator.clusterserviceversion.yaml]: Done

File: config/manifests/openshift/kustomization.yaml (hunk 1 above)
Comment (rollandf) [reply config/manifests/openshift/kustomization.yaml]: Done

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**